### PR TITLE
[#1679] Split and refactor device registry module.

### PR DESCRIPTION
**Pull request by kaniyan**
Currently I am working on the issue #1679, which is implementing a _Mongodb_ based device-registry. IMHO the existing device-registry module could be split into various sub-modules and also the classes could be reorganised under different packages. I think this enables to understand the code better and also to easier to maintain. This PR is a proposal how it could be refactored.

 In the existing device-registry module, although there is a possibility to add more implementations (eg. _Mongodb_, _Infinispan_ based etc.) by enabling the spring _services/repositories_ based on the conditional property, this approach bundles all the the implementation classes and the dependencies into a single jar/image. With the propsal in this PR, this can be avoided and each device-registry implementations could be created into a separate jar/image. 

The helm charts could also be adapted in such a way to specify which implementation ie. a _File_ or _Mongodb_  or _Infinispan_ based device-registry should be deployed.

### CODE REVIEW

**Comment by calohmn** [comment]
With these changes there is now a
`services/device-registry` module with
`device-registry-base` and
`device-registry-file`
as submodules of that.

My intuition here would be to rather have the extra `device-registry` layer removed, so that we end up with
`services/device-registry-base` and
`services/device-registry-file`.
WDYT?

**Comment by kaniyan** [comment]
My idea was to group the _base_ module and other device registry implementations under the `device-registry` module. Its also fine for me to move them under `services` and remove the `device-registry` module.

File: services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/deviceConnection/AutowiredDeviceConnectionAdapter.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
  * SPDX-License-Identifier: EPL-2.0
  *******************************************************************************/
 
-package org.eclipse.hono.deviceregistry;
+package org.eclipse.hono.deviceregistry.service.deviceConnection;
```
Comment (calohmn) [inline services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/deviceConnection/AutowiredDeviceConnectionAdapter.java]: Package name should be all lowercase, so `deviceconnection`.

File: services/device-registry-base/src/test/java/org/eclipse/hono/deviceregistry/service/deviceConnection/MapBasedDeviceConnectionServiceTest.java [hunk 2]
```diff
@@ -11,7 +11,7 @@
  * SPDX-License-Identifier: EPL-2.0
  *******************************************************************************/
 
-package org.eclipse.hono.deviceregistry;
+package org.eclipse.hono.deviceregistry.service.deviceConnection;
```
Comment (calohmn) [inline services/device-registry-base/src/test/java/org/eclipse/hono/deviceregistry/service/deviceConnection/MapBasedDeviceConnectionServiceTest.java]: Package name all lowercase (`deviceconnection`) also in `test` directory.

File: services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/deviceConnection/AutowiredDeviceConnectionAdapter.java (hunk 1 above)
Comment (kaniyan) [reply services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/deviceConnection/AutowiredDeviceConnectionAdapter.java]: My bad. I have changed it now.

**Pull request merged**